### ISSUE TITLE
android build failed

### ISSUE THREAD

**Issue by tsao-chi** [minetest/minetest#9590]
##### Minetest version
https://github.com/minetest/minetest/commit/b8d29ab0aff9fd141e9b59e941c47f4b2d3d5bfc
##### Summary
android build failed

##### Steps to reproduce
use the script of https://github.com/The-Unstable-World/The-Unstable-World to build

##### logs:
https://travis-ci.com/github/The-Unstable-World/The-Unstable-World/jobs/313869188
https://travis-ci.com/github/The-Unstable-World/The-Unstable-World/jobs/313406387
```
404 Not Found
2020-04-04 13:03:47 ERROR 404: Not Found.

Makefile:280: recipe for target 'openssl_download' failed
make[1]: *** [openssl_download] Error 1
make[1]: *** Waiting for unfinished jobs....
```
latest successful build: https://travis-ci.com/github/The-Unstable-World/The-Unstable-World/jobs/311915764

**Comment by SmallJoker** [comment]
Does this fix your issue? #9066

**Comment by juozaspo** [comment]
Your script tries to download an old version of openssl which is no longer available (`1.0.2n`) at the primary location of the download page (`/source`). It has been moved to `/source/old/1.0.2`. The latest version is `1.1.1f`. Excerpt(s) from openssl download page:

> All releases can be found at /source/old

> The latest stable version is the 1.1.1 series. This is also our Long Term Support (LTS) version, supported until 11th September 2023. All other versions (including 1.1.0, 1.0.2, 1.0.0 and 0.9.8) are now out of support and should not be used. Users of these older versions are encourage to upgrade to 1.1.1 as soon as possible. Extended support for 1.0.2 to gain access to security fixes for that version is available.

The information in `build/android/Makefile` (and possibly other files that I didn't check) needs to be updated and build needs to be tested.
You can either change openssl version to `1.1.1f` in that file or change the openssl download url to include `/old/1.0.2` for version 1.0.2. E.g. `https://www.openssl.org/source/old/1.0.2/openssl-1.0.2n.tar.gz` or `https://www.openssl.org/source/old/1.0.2/openssl-$(OPENSSL_VERSION).tar.gz`
